packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/components/SubjectGroupSubTaskPopup.js

```diff
@@ -0,0 +1,285 @@
+      if (TaskModel) {
+        const taskKey = `${currentTask.taskKey}.cell.${cellIndex}.subtask.${index}`
+
+        const taskSnapshot = {
+          taskKey,
+          type: subtaskDef.type,
+          required: subtaskDef.required || false,
+          strings: {
+            question: subtaskDef.question || subtaskDef.instruction || 'Select an option',
+            instruction: subtaskDef.instruction || subtaskDef.question || '',
+            help: subtaskDef.help || ''
+          }
+        }
+
+        if (subtaskDef.answers && Array.isArray(subtaskDef.answers)) {
+          taskSnapshot.answers = subtaskDef.answers.map(answer => {
+            if (typeof answer === 'string') {
+              return { label: answer }
+            } else if (answer && typeof answer === 'object') {
+              return {
+                label: answer.label || answer.value || String(answer)
+              }
+            }
+            return { label: String(answer) }
+          }).filter(answer => answer.label)
+
+          subtaskDef.answers.forEach((answer, answerIndex) => {
+            const answerLabel = typeof answer === 'string' ? answer : (answer?.label || answer?.value || String(answer))
+            taskSnapshot.strings[`answers.${answerIndex}.label`] = answerLabel
+          })
+        }
+
+        const task = TaskModel.create(taskSnapshot)
+        return task
+      }
+    } catch (error) {
+      console.warn(`Could not create Task for subtask type ${subtaskDef.type}:`, error)
+    }
+    return null
+  }).filter(Boolean)
+
+  function getAnnotationForTask(task) {
+    return currentCell.annotation(task)
+  }
+
+  function createAnnotationForTask(task) {
+    return currentCell.addAnnotation(task)
+  }
+
+  function isTaskComplete(task) {
+    const annotation = getAnnotationForTask(task)
+    if (!annotation) return false
+    return annotation.isComplete
+  }
+
+  function areAllRequiredTasksComplete() {
+    return subtasks.every(subtask => {
+      const isRequired = subtask.required === 'true' || subtask.required === true
+      return !isRequired || isTaskComplete(subtask)
+    })
+  }
+
+  function handleClose(event) {
+    event?.preventDefault()
+    event?.stopPropagation()
+    if (!areAllRequiredTasksComplete()) {
+      setConfirm('confirming')
+    } else {
+      onSave({})
+      onClose()
+    }
+  }
+
+  function onCloseConfirm() {
+    setConfirm('closed')
+  }
+
+  function deleteCellData() {
+    setConfirm('closed')
+    subtasks.forEach(subtask => {
+      const annotation = getAnnotationForTask(subtask)
+      if (annotation) {
+        currentCell.removeAnnotation(annotation)
+      }
+    })
+    onSave({})
+    onClose()
+  }
+
+  function handleCancelSelection() {
+    subtasks.forEach(subtask => {
+      const annotation = getAnnotationForTask(subtask)
+      if (annotation) {
+        currentCell.removeAnnotation(annotation)
+      }
+    })
+    if (onCancelSelection) {
+      onCancelSelection()
+    }
```

Comment:
[minor] This can be simplified to `onCancelSelection?.()`, or if we have a default handler for onCancelSelection, we don't even need to worry about that!